scalapb-runtime/src/main/scala/scalapb/descriptors/Descriptor.scala

```diff
@@ -149,34 +149,30 @@ class MethodDescriptor private[descriptors] (
 
   def getOptions = asProto.getOptions
 
-  lazy val inputType: ScalaType = {
+  lazy val inputType: Descriptor = {
     val inputType = asProto.inputType.getOrElse(
       throw new DescriptorValidationException(this, "missing method input type")
     )
 
-    ScalaType.Message(
-      FileDescriptor
-        .find(containingService.file, this, inputType)
-        .getOrElse(
-          throw new DescriptorValidationException(this, "couldn't build input type descriptor")
-        )
-        .asInstanceOf[Descriptor]
-    )
+    FileDescriptor
+      .find(containingService.file, this, inputType)
+      .getOrElse(
```

Comment:
lets use pattern matching on `case Some(d: Descriptor)` to make the asInstanceOf unnecessary and handle the case that what's being found isn't a descriptor.